posthog/hogql_queries/experiments/experiment_trends_statistics.py

```diff
@@ -0,0 +1,204 @@
+    if len(test_variants) < 1:
+        raise ValidationError("Can't calculate experiment results for less than 2 variants", code="no_data")
+
+    # Calculate posterior parameters for control
+    alpha_control = PRIOR_ALPHA + control_variant.count
+    beta_control = PRIOR_BETA + control_variant.exposure
+
+    # Draw samples from control posterior
+    samples_control = gamma.rvs(alpha_control, scale=1 / beta_control, size=SAMPLE_SIZE)
+
+    # Draw samples for each test variant
+    test_samples = []
+    for test in test_variants:
+        alpha_test = PRIOR_ALPHA + test.count
+        beta_test = PRIOR_BETA + test.exposure
+        test_samples.append(gamma.rvs(alpha_test, scale=1 / beta_test, size=SAMPLE_SIZE))
+
+    # Calculate probabilities
+    probabilities = []
+
+    # Probability control wins (beats all test variants)
+    control_wins = np.all([samples_control > test_sample for test_sample in test_samples], axis=0)
+    probabilities.append(float(np.mean(control_wins)))
+
+    # Probability each test variant wins (beats control and all other test variants)
+    for i, test_sample in enumerate(test_samples):
+        other_test_samples = test_samples[:i] + test_samples[i + 1 :]
+        variant_wins = np.all(
+            [test_sample > samples_control] + [test_sample > other for other in other_test_samples], axis=0
+        )
+        probabilities.append(float(np.mean(variant_wins)))
+
+    return probabilities
+
+
+def are_results_significant(
+    control_variant: ExperimentVariantTrendsBaseStats,
+    test_variants: list[ExperimentVariantTrendsBaseStats],
+    probabilities: list[Probability],
+) -> tuple[ExperimentSignificanceCode, Probability]:
+    """
+    Determines if experiment results are statistically significant using Bayesian analysis.
+
+    This function evaluates the win probabilities of each variant to determine if any variant
+    is significantly better than the others. The method:
+    1. Checks if sample sizes meet minimum threshold requirements
+    2. Evaluates win probabilities from the posterior distributions
+
+    Parameters:
+    -----------
+    control_variant : ExperimentVariantTrendsBaseStats
+        Statistics for the control group, including count and exposure data
+    test_variants : list[ExperimentVariantTrendsBaseStats]
+        List of statistics for test variants to compare against control
+    probabilities : list[Probability]
+        List of win probabilities for each variant, as calculated by calculate_probabilities
+
+    Returns:
+    --------
+    tuple[ExperimentSignificanceCode, Probability]
+        - ExperimentSignificanceCode indicating the significance status:
+            NOT_ENOUGH_EXPOSURE: Insufficient sample size
+            LOW_WIN_PROBABILITY: No variant has a high enough probability of being best
+            SIGNIFICANT: Clear winner with high probability of being best
```

Comment:
I've realized there's something missing in my cleaned up notebook.

There's one additional criterion that must be met before we can declare results significant. In the current implementation:
- For Trends, the p value must be below certain threshold
- For Funnels, the "expected loss" value must be below a certain threshold.

In the new implementation, we'll be calculating "expected loss" for both Trends and Funnels, as using p values in not appropriate.

Because this criterion is missing in your implementation, we might get more instances of significant results than in the old method.

I need to look at the stats methods once more to see how this is calculated. We did discuss this with Jimmy and should be a fairly simple additional check. Not blocking for now, can add this in a follow up.